lib/ex_unit/lib/ex_unit/capture_io.ex

```diff
@@ -110,30 +110,29 @@ defmodule ExUnit.CaptureIO do
   end
 
   defp do_capture_io(device, options, fun) do
-    unless original_io = Process.whereis(device) do
-      raise "could not find IO device registered at #{inspect device}"
-    end
-
-    unless ExUnit.Server.add_device(device) do
-      raise "IO device registered at #{inspect device} is already captured"
-    end
-
     input = Keyword.get(options, :input, "")
-
-    Process.unregister(device)
-    {:ok, capture_io} = StringIO.open(input)
-    Process.register(capture_io, device)
+    {:ok, string_io} = StringIO.open(input)
+    {ref, pid} = case ExUnit.Server.capture_device(device, string_io) do
+      {:ok, ref, pid} -> {ref, pid}
+      {:error, :no_device} ->
+        _ = StringIO.close(string_io)
+        raise "could not find IO device registered at #{inspect device}"
+      {:error, :already_captured} ->
+        _ = StringIO.close(string_io)
+        raise "IO device registered at #{inspect device} is already captured"
+    end
 
     try do
-      do_capture_io(capture_io, fun)
+      Process.unregister(device)
+      Process.register(pid, device)
```

Comment:
Definitely it should, it's copied verbatim here to see how it looks like on client.